chaoscenter/authentication/pkg/project/repository.go

```diff
@@ -69,51 +70,148 @@ func (r repository) GetProjects(query bson.D) ([]*entities.Project, error) {
 }
 
 // GetProjectsByUserID returns a project based on the userID
-func (r repository) GetProjectsByUserID(userID string, isOwner bool) ([]*entities.Project, error) {
+func (r repository) GetProjectsByUserID(request *entities.ListProjectRequest) (*entities.ListProjectResponse, error) {
 	var projects []*entities.Project
-	query := bson.D{}
 
-	if isOwner == true {
-		query = bson.D{
-			{"members", bson.D{
-				{"$elemMatch", bson.D{
-					{"user_id", userID},
-					{"role", bson.D{
-						{"$eq", entities.RoleOwner},
+	// Construct the pipeline
+	var pipeline mongo.Pipeline
```

Comment:
create util functions for match, filter and sort stages. This function is way too long

Reply:
done